### ISSUE TITLE
Ensure that `redis::AsyncIter` didn't get error while retrieving values by SCAN cmd

### ISSUE THREAD

**Issue by alefbragin** [redis-rs/redis-rs#1309]
Currently, `redis::AsyncIter` yields just type `T` rather than `Result<T>`, and there is no ability to check possible errors during long SCAN process

**Comment by nihohit** [comment]
Do you think this can be done in a backwards-compatible way?

**Comment by threema-donat** [comment]
I think it could be done in a backwards-compatible way, but it would mean a lot of work without that much of benefit. The `Iter` and `AsyncIter` would have to be duplicated with a fix. Furthermore, all functions that return some sort of iterator would have to be duplicated (e.g. `try_scan`).

I would be very motivated to open a PR that will make the iterators yield a `RedisResult<T>`, but not so much in doing it in a backwards compatible way. My reasoning is that it is better when it breaks the current implementations since they rely on the conversion always being successful, which can lead to a lot of problems. Also, the library is `<1.0`.

Would you be willing to accept such a PR?

**Comment by nihohit** [comment]
I agree with the sentiment, and this [breaking change is planned](https://github.com/redis-rs/redis-rs/issues/1043), but ATM there's no plan or timetable for version 1.0.0, so I prefer not to break API.

**Comment by nihohit** [comment]
fixed in https://github.com/redis-rs/redis-rs/pull/1641
